In [1]:
import pandas as pd
import numpy as np

In [2]:
import altair as alt

In [3]:
def tab_reader(filepathpattern,tabnum,skiprow,low_memory=False):
    tabs = []
    for i in range (tabnum):
        cur_index = i+1
        cur_filepath = filepathpattern.format(cur_index)
        try:
            tab = pd.read_csv(cur_filepath, skiprows=skiprow, low_memory=False)
        except:
            tab = pd.read_csv(cur_filepath, skiprows=skiprow,encoding='latin-1', low_memory=low_memory)
        tabs.append(tab)
    return tabs


In [4]:
# 2017 data
rcbp_2017_mb_profitMargin_Tabs = tab_reader("../../../data/2017_csv_eng/2017_Medium businesses_Profit margin based_csv/_2017_Medium businesses_Profit margin_Tab{}.csv",5,5)
rcbp_2017_mb_revenue_Tabs = tab_reader("../../../data/2017_csv_eng/2017_Medium businesses_Total revenue based_csv/_2017_Eng_Medium_Revenue_Tab{}.csv",5,5)
rcbp_2017_sb_profitMargin_Tabs = tab_reader("../../../data/2017_csv_eng/2017_Small businesses_Profit margin based_csv/_2017_Small businesses_Profit margin based_Tab{}.csv",7,5)
rcbp_2017_sb_profitMargin_Tabs[0] = tab_reader("../../../data/2017_csv_eng/2017_Small businesses_Profit margin based_csv/_2017_Small businesses_Profit margin based_Tab{}.csv",1,6)[0]
rcbp_2017_sb_revenue_Tabs = tab_reader("../../../data/2017_csv_eng/2017_Small businesses_Total revenue based_csv/_2017_Small businesses_Total revenue based_Tab{}.csv",7,5)
rcbp_2017_sb_revenue_Tabs[0] = tab_reader("../../../data/2017_csv_eng/2017_Small businesses_Total revenue based_csv/_2017_Small businesses_Total revenue based_Tab{}.csv",1,6)[0]

In [5]:
# 2018 data
rcbp_2018_mb_profitMargin_Tabs = tab_reader("../../../data/2018_csv_eng/2018_Medium businesses_Profit margin based_csv/_2018_Medium businesses_Profit margin_Tab{}.csv",5,5)
rcbp_2018_mb_revenue_Tabs = tab_reader("../../../data/2018_csv_eng/2018_Medium businesses_Total revenue based_csv/_2018_Medium businesses_Total revenue_Tab{}.csv",5,5)
rcbp_2018_sb_profitMargin_Tabs = tab_reader("../../../data/2018_csv_eng/2018_Small businesses_Profit margin based_csv/_2018_Small businesses_Profit margin_Tab{}.csv",7,5)
rcbp_2018_sb_profitMargin_Tabs[0] = tab_reader("../../../data/2018_csv_eng/2018_Small businesses_Profit margin based_csv/_2018_Small businesses_Profit margin_Tab{}.csv",1,6)[0]
rcbp_2018_sb_revenue_Tabs = tab_reader("../../../data/2018_csv_eng/2018_Small businesses_Total revenue based_csv/_2018_Small businesses_Total revenue_Tab{}.csv",7,5)
rcbp_2018_sb_revenue_Tabs[0] = tab_reader("../../../data/2018_csv_eng/2018_Small businesses_Total revenue based_csv/_2018_Small businesses_Total revenue_Tab{}.csv",1,6)[0]

In [6]:
# 2019 data
rcbp_2019_mb_profitMargin_Tabs = tab_reader("../../../data/2019_csv_eng/2019_Medium businesses_Profit margin_csv/_2019_Medium businesses_Profit margin_Tab{}.csv",5,5)
rcbp_2019_mb_revenue_Tabs = tab_reader("../../../data/2019_csv_eng/2019_Medium businesses_Total revenue_csv/_2019_Medium businesses_Total revenue_Tab{}.csv",5,5)
rcbp_2019_sb_profitMargin_Tabs = tab_reader("../../../data/2019_csv_eng/2019_Small businesses_Profit margin_csv/_2019_Small businesses_Profit margin_Tab{}.csv",7,5)
rcbp_2019_sb_profitMargin_Tabs[0] = tab_reader("../../../data/2019_csv_eng/2019_Small businesses_Profit margin_csv/_2019_Small businesses_Profit margin_Tab{}.csv",1,6)[0]
rcbp_2019_sb_revenue_Tabs = tab_reader("../../../data/2019_csv_eng/2019_Small businesses_Total revenue_csv/_2019_Small businesses_Total revenue_Tab{}.csv",7,5)
rcbp_2019_sb_revenue_Tabs[0] = tab_reader("../../../data/2019_csv_eng/2019_Small businesses_Total revenue_csv/_2019_Small businesses_Total revenue_Tab{}.csv",1,6)[0]

In [7]:
def selectData(tabs):
    concatTabs = []
    for tab in tabs:
        cols_without_quartile = [x for x in tab.columns if 'quartile' not in x.lower()]
        tab = tab[cols_without_quartile]
        tab = tab.loc[(tab["North American Industry Classification System, NAICS - code"].str.len() < 3) | (tab["North American Industry Classification System, NAICS - code"].str.contains('-'))]
        concatTabs.append(tab)
    return pd.concat(concatTabs)

In [8]:
mb_2017_revenue_tab2 = rcbp_2017_mb_revenue_Tabs[1]
mb_2018_revenue_tab2 = rcbp_2018_mb_revenue_Tabs[1]
mb_2019_revenue_tab2 = rcbp_2019_mb_revenue_Tabs[1]
mb_revenue_persent_3y = selectData([mb_2017_revenue_tab2,mb_2018_revenue_tab2,mb_2019_revenue_tab2])

In [9]:
mb_2017_revenue_tab3 = rcbp_2017_mb_revenue_Tabs[2]
mb_2018_revenue_tab3 = rcbp_2018_mb_revenue_Tabs[2]
mb_2019_revenue_tab3 = rcbp_2019_mb_revenue_Tabs[2]
mb_expenses_value_3y = selectData([mb_2017_revenue_tab3,mb_2018_revenue_tab3,mb_2019_revenue_tab3])

In [10]:
sb_2017_revenue_tab2 = rcbp_2017_sb_revenue_Tabs[1]
sb_2018_revenue_tab2 = rcbp_2018_sb_revenue_Tabs[1]
sb_2019_revenue_tab2 = rcbp_2019_sb_revenue_Tabs[1]
sb_revenue_persent_3y = selectData([sb_2017_revenue_tab2,sb_2018_revenue_tab2,sb_2019_revenue_tab2])

In [11]:
sb_2017_revenue_tab3 = rcbp_2017_sb_revenue_Tabs[2]
sb_2018_revenue_tab3 = rcbp_2018_sb_revenue_Tabs[2]
sb_2019_revenue_tab3 = rcbp_2019_sb_revenue_Tabs[2]
sb_revenue_value_3y = selectData([sb_2017_revenue_tab3,sb_2018_revenue_tab3,sb_2019_revenue_tab3])

# Medium businesses

### Total expenses - All industry - Medium businesses

In [12]:
# chart 1 - total expenses - all industry
chart1_df = mb_expenses_value_3y[(mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_expenses_value_3y['Total expenses, percent of businesses reporting'].astype(float) >= 90) & (mb_expenses_value_3y['Total expenses, quality indicator'] < 'E')]
chart1_df = chart1_df[['Reference year','Location indicator','Total expenses']]

In [13]:
# chart 1_1 - total expenses - all industry
chart1_1_df = mb_revenue_persent_3y[(mb_revenue_persent_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_revenue_persent_3y['Total expenses (%), percent of businesses reporting'].astype(float) >= 90) & (mb_revenue_persent_3y['Total expenses (%), quality indicator'] < 'E')]
chart1_1_df = chart1_1_df[['Reference year','Location indicator','Total expenses (%)']]

In [14]:
chart1 = alt.Chart(chart1_df, title=alt.TitleParams(text = "Average total expenses of medium businesses", 
                                           subtitle = "Canada, all inductries, by rural and urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('Location indicator', title=''),
    y=alt.Y('Total expenses:Q', title='Total Expenses'),
    color=alt.Color('Location indicator', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [15]:
chart1_1 = alt.Chart(chart1_1_df, title=alt.TitleParams(text = "Percentage of total expenses in total revenues of medium businesses", 
                                           subtitle = "Canada, all inductries, by rural and urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('Location indicator', title=''),
    y=alt.Y('Total expenses (%):Q', title='Total Expenses (%)'),
    color=alt.Color('Location indicator', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [16]:
chart1 | chart1_1

alt.HConcatChart(...)

In [17]:
pd.merge(left=chart1_df[chart1_df['Location indicator'] == 'Rural'],right=chart1_df[chart1_df['Location indicator'] == 'Urban'],how='inner',on='Reference year')

Reference year Location indicator_x Total expenses_x Location indicator_y  \
0          2017.0                Rural           8896.5                Urban   
1          2018.0                Rural           8907.5                Urban   
2          2019.0                Rural           8973.3                Urban   

  Total expenses_y  
0           8893.6  
1           8844.4  
2             8873

In [18]:

pd.merge(left=chart1_1_df[chart1_1_df['Location indicator'] == 'Rural'],right=chart1_1_df[chart1_1_df['Location indicator'] == 'Urban'],how='inner',on='Reference year')

Reference year Location indicator_x Total expenses (%)_x  \
0          2017.0                Rural                 96.1   
1          2018.0                Rural                 96.2   
2          2019.0                Rural                 96.4   

  Location indicator_y Total expenses (%)_y  
0                Urban                 93.6  
1                Urban                 92.9  
2                Urban                 92.9

**The average total expenses of medium businesses in both rural and urban area did not have significant changes in the three years from 2017 to 2019. The average total expenses for rural medium businesses increased by 76.8 from 8896.5K in 2017 to 8973.3K in 2019, while for urban medium businesses, this value increased by 76.8K from 8896.5K in 2017 to 8973.3K in 2019, while for urban medium businesses, this value decreased by 20.6K from 8893.6K in 2017 to 8873K in 2019. Compared with urban businesses, rural businesses have higher total expenses as a percentage of their total revenues. In addition, the total expenses percentage of rural businesses increased 0.3%, while the total expenses percentage of urban businesses decreased 0.7%. Although there are slight changes on total expenses value, the urban businesses seems had slightly better performance on expenses control.**


### Rural/Urban direct/indirect cost expenses - all industry - medium businesses

In [19]:
chart3_df = mb_expenses_value_3y[(mb_expenses_value_3y['Location indicator'] == 'Rural') & (mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_expenses_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (mb_expenses_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart3_df = chart3_df[['Reference year','Location indicator','Cost of sales (direct expenses)','Operating expenses (indirect expenses)']]
chart3_cp = chart3_df

In [20]:
chart3_df = chart3_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [21]:
chart3 = alt.Chart(chart3_df,title=alt.TitleParams(text = "Average direct/indirect expenses of medium businesses", 
                                           subtitle = "Canada, all inductries, rural areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('type', title=''),
    y=alt.Y('value:Q', title='Expenses Value'),
    color=alt.Color('type', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [22]:
chart4_df = mb_expenses_value_3y[(mb_expenses_value_3y['Location indicator'] == 'Urban') & (mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_expenses_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (mb_expenses_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart4_df = chart4_df[['Reference year','Location indicator','Cost of sales (direct expenses)','Operating expenses (indirect expenses)']]
chart4_cp = chart4_df

In [23]:
chart4_df = chart4_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [24]:
chart4 = alt.Chart(chart4_df, title=alt.TitleParams(text = "Average direct/indirect expenses of medium businesses", 
                                           subtitle = "Canada, all inductries, urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('type', title=''),
    y=alt.Y('value:Q', title='Expenses Value'),
    color=alt.Color('type', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [25]:
chart3 | chart4

alt.HConcatChart(...)

In [26]:
chart3_df

Reference year Location indicator                                    type  \
0          2017.0              Rural         Cost of sales (direct expenses)   
1          2018.0              Rural         Cost of sales (direct expenses)   
2          2019.0              Rural         Cost of sales (direct expenses)   
3          2017.0              Rural  Operating expenses (indirect expenses)   
4          2018.0              Rural  Operating expenses (indirect expenses)   
5          2019.0              Rural  Operating expenses (indirect expenses)   

    value  
0  5418.4  
1  5412.4  
2  5402.4  
3  3478.1  
4  3495.1  
5  3570.9

In [27]:
chart3_cp['Cost of sales (direct expenses)'].astype(float).mean()

5411.066666666667

In [28]:
chart3_cp['Operating expenses (indirect expenses)'].astype(float).mean()

3514.7000000000003

In [29]:
chart4_df

Reference year Location indicator                                    type  \
0          2017.0              Urban         Cost of sales (direct expenses)   
1          2018.0              Urban         Cost of sales (direct expenses)   
2          2019.0              Urban         Cost of sales (direct expenses)   
3          2017.0              Urban  Operating expenses (indirect expenses)   
4          2018.0              Urban  Operating expenses (indirect expenses)   
5          2019.0              Urban  Operating expenses (indirect expenses)   

    value  
0  5635.2  
1  5578.7  
2  5537.5  
3  3258.4  
4  3265.7  
5  3335.5

In [30]:
chart4_cp['Cost of sales (direct expenses)'].astype(float).mean()

5583.8

In [31]:
chart4_cp['Operating expenses (indirect expenses)'].astype(float).mean()

3286.5333333333333

**For medium rural businesses, in recorded 3 years, the average direct expenses was 5411.06K which was larger than the average indirect expenses (3514.7K). For medium urban businesses, between 2017 and 2019, the average direct expenses (5583.8K) was also larger than the average indirect expenses (3286.5K). For direct expenses, average expenses values of medium businesses in rural area was lower than urban area, while for indirect expenses, average expenses values of medium businesses in rural area was higher than urban area.**
 

### Rural/Urban different type cost expenses - all industry - medium businesses - all businesses

In [32]:
chart13_df = mb_expenses_value_3y[(mb_expenses_value_3y['Location indicator'] == 'Rural') & (mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_expenses_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (mb_expenses_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart13_df = chart13_df[['Reference year','Location indicator',
                        'Wages and benefits', 'Purchases, materials and sub-contracts', 'Opening inventory', 
                           'Closing inventory', 'Labour and commissions', 'Amortization and depletion',
                           'Repairs and maintenance', 'Utilities and telephone/telecommunication',
                         'Rent', 'Interest and bank charges', 'Professional and business fees',
                         'Advertising and promotion', 'Delivery, shipping and warehouse', 'Insurance', 
                         'Other indirect expenses']]
chart13_cp = chart13_df

In [33]:
chart13_df = chart13_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [34]:
chart13_df['value'] = chart13_df['value'].astype(float)

In [35]:
chart13 = alt.Chart(chart13_df,title=alt.TitleParams(text = "Distribution of total expenses", 
                                           subtitle = "Canada, all inductries, medium businesses, rural areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('value:Q', title='Expenses Value'),
    y=alt.Y('type', title='',sort='x'),
    color=alt.Color('type', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=400, height=300)

In [36]:
chart14_df = mb_expenses_value_3y[(mb_expenses_value_3y['Location indicator'] == 'Urban') & (mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (mb_expenses_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (mb_expenses_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart14_df = chart14_df[['Reference year','Location indicator',
                        'Wages and benefits', 'Purchases, materials and sub-contracts', 'Opening inventory', 
                           'Closing inventory', 'Labour and commissions', 'Amortization and depletion',
                           'Repairs and maintenance', 'Utilities and telephone/telecommunication',
                         'Rent', 'Interest and bank charges', 'Professional and business fees',
                         'Advertising and promotion', 'Delivery, shipping and warehouse', 'Insurance', 
                         'Other indirect expenses']]
chart14_cp = chart14_df

In [37]:
chart14_df = chart14_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [38]:
chart14_df['value'] = chart14_df['value'].astype(float)

In [39]:
chart14 = alt.Chart(chart14_df,title=alt.TitleParams(text = "Distribution of total expenses", 
                                           subtitle = "Canada, all inductries, medium businesses, urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('value:Q', title='Expenses Value'),
    y=alt.Y('type', title='',sort='x'),
    color=alt.Color('type', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=400, height=300)

In [40]:
chart13 | chart14

alt.HConcatChart(...)

In [41]:
chart13_cp

Reference year Location indicator Wages and benefits  \
1          2017.0              Rural              598.2   
1          2018.0              Rural              616.4   
1          2019.0              Rural              642.3   

  Purchases, materials and sub-contracts Opening inventory Closing inventory  \
1                                 4889.5             911.2             980.4   
1                                 4875.9             899.3             979.2   
1                                 4801.2             941.6             982.6   

  Labour and commissions Amortization and depletion Repairs and maintenance  \
1                    983                      266.6                   183.1   
1                  976.7                      261.5                   185.4   
1                  998.4                      278.3                     187   

  Utilities and telephone/telecommunication   Rent Interest and bank charges  \
1                                       142    187                        83   
1                                     150.3  184.1                      88.9   
1                                     149.4  188.9                      97.2   

  Professional and business fees Advertising and promotion  \
1                           89.3                      63.7   
1                           86.4                      61.3   
1                           85.2                      63.2   

  Delivery, shipping and warehouse Insurance Other indirect expenses  
1                             49.4      57.3                  1373.8  
1                             48.9        58                  1393.5  
1                             51.4      63.1                  1408.9

In [42]:
total = chart13_df[chart13_df['Reference year'] == 2019]['value'].astype(float).sum()
print('Purchases, materials and sub-contracts:', float(chart13_cp[chart13_cp['Reference year'] == 2019].iloc[0]['Purchases, materials and sub-contracts'])/total)
print('Other indirect expenses:', float(chart13_cp[chart13_cp['Reference year'] == 2019].iloc[0]['Other indirect expenses'])/total)
print('Labour and commissions:', float(chart13_cp[chart13_cp['Reference year'] == 2019].iloc[0]['Labour and commissions'])/total)
print('Closing inventory:', float(chart13_cp[chart13_cp['Reference year'] == 2019].iloc[0]['Closing inventory'])/total)
print('Opening inventory:', float(chart13_cp[chart13_cp['Reference year'] == 2019].iloc[0]['Opening inventory'])/total)

Purchases, materials and sub-contracts: 0.43891870149103634
Other indirect expenses: 0.12879958313145073
Labour and commissions: 0.09127227184217503
Closing inventory: 0.08982785888633932
Opening inventory: 0.08607969868448719


In [43]:
chart14_cp

Reference year Location indicator Wages and benefits  \
2          2017.0              Urban              805.5   
2          2018.0              Urban              821.2   
2          2019.0              Urban              827.9   

  Purchases, materials and sub-contracts Opening inventory Closing inventory  \
2                                 4862.4             886.3             918.9   
2                                 4825.3               853             920.7   
2                                 4731.8             872.2             894.4   

  Labour and commissions Amortization and depletion Repairs and maintenance  \
2                 1303.9                      214.4                    93.3   
2                 1316.3                      208.4                    96.4   
2                 1340.4                      220.6                    98.1   

  Utilities and telephone/telecommunication   Rent Interest and bank charges  \
2                                      91.2  221.1                     106.8   
2                                      92.3  223.5                     118.5   
2                                      93.4    228                     127.6   

  Professional and business fees Advertising and promotion  \
2                          172.9                     109.2   
2                          174.5                     108.2   
2                          177.4                     113.2   

  Delivery, shipping and warehouse Insurance Other indirect expenses  
2                             58.4      44.2                   843.2  
2                             54.8      44.3                   828.5  
2                               59      48.2                   829.4

In [44]:
total = chart14_df[chart14_df['Reference year'] == 2019]['value'].astype(float).sum()
print('Purchases, materials and sub-contracts:', float(chart14_cp[chart14_cp['Reference year'] == 2019].iloc[0]['Purchases, materials and sub-contracts'])/total)
print('Other indirect expenses:', float(chart14_cp[chart14_cp['Reference year'] == 2019].iloc[0]['Other indirect expenses'])/total)
print('Labour and commissions:', float(chart14_cp[chart14_cp['Reference year'] == 2019].iloc[0]['Labour and commissions'])/total)
print('Closing inventory:', float(chart14_cp[chart14_cp['Reference year'] == 2019].iloc[0]['Closing inventory'])/total)
print('Opening inventory:', float(chart14_cp[chart14_cp['Reference year'] == 2019].iloc[0]['Opening inventory'])/total)

Purchases, materials and sub-contracts: 0.4438170631049749
Other indirect expenses: 0.07779320177084115
Labour and commissions: 0.12572221805357545
Closing inventory: 0.08388984767764687
Opening inventory: 0.0818076086140917


**The distribution of medium business expenses among expenses types was similar in the three years from 2017 to 2019, but it was different in rural and urban area.**

**In 2019, rural area, 'Purchases, materials and sub-contracts' (43.89%), 'Other indirect expenses' (12.88%) were top 2 expenses types, which accounted for more than half (56.77%) of rural medium business expenses. For urban area, 'Purchases, materials and sub-contracts' (44.38%), 'Labour and commissions' (12.57%) were top 2 expenses types that accounted 56.95% expenses in total. Compared to the rural businesses, urban businesses speend more expenses on 'Labour and commissions'.**

**In 2019, regardless of the location of businesses (both rural and urban), the opening inventory smaller than closing inventory. This is a bad indicator, which might mean that the produce was more than sell.**

### industry shares of total expenses - rural area

In [45]:
# chart 15 - total expenses - all industry
chart15_df = mb_expenses_value_3y[(mb_expenses_value_3y['North American Industry Classification System, NAICS - code'] != '0') & (mb_expenses_value_3y['Total expenses, percent of businesses reporting'].astype(float) >= 90) & (mb_expenses_value_3y['Total expenses, quality indicator'] < 'E')]
chart15_df = chart15_df[['North American Industry Classification System, NAICS','Reference year','Location indicator','Total expenses']]

In [46]:
chart15_df = chart15_df[(chart15_df['Location indicator'] == 'Rural') & (chart15_df['Reference year'] == 2019)]

In [47]:
chart15_df['Total expenses'] = chart15_df['Total expenses'].astype(float)

In [48]:
alt.Chart(chart15_df,title=alt.TitleParams(text = "Industry shares of total expenses", 
                                           subtitle = "Canada, medium businesses, urban areas, 2019")).mark_bar().encode(
    x=alt.X('Total expenses:Q', title='Total Expenses'),
    y=alt.Y('North American Industry Classification System, NAICS', title='',sort='x'),
    color=alt.Color('North American Industry Classification System, NAICS', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=400, height=300)

alt.Chart(...)

In [49]:
chart15_df.sort_values(['Total expenses'])

North American Industry Classification System, NAICS  Reference year  \
175            Management of companies and enterprises            2019.0   
185                               Educational services            2019.0   
171    Professional, scientific and technical services            2019.0   
163                 Real estate and rental and leasing            2019.0   
213      Other services (except public administration)            2019.0   
149                Information and cultural industries            2019.0   
179  Administrative and support, waste management a...            2019.0   
207                    Accommodation and food services            2019.0   
189                  Health care and social assistance            2019.0   
199                 Arts, entertainment and recreation            2019.0   
125                     Transportation and warehousing            2019.0   
27                                        Construction            2019.0   
15       Mining, quarrying, and oil and gas extraction            2019.0   
23                                           Utilities            2019.0   
35                                       Manufacturing            2019.0   
3           Agriculture, forestry, fishing and hunting            2019.0   
99                                        Retail trade            2019.0   
79                                     Wholesale trade            2019.0   

    Location indicator  Total expenses  
175              Rural          4287.4  
185              Rural          6482.0  
171              Rural          6577.3  
163              Rural          6882.6  
213              Rural          7646.5  
149              Rural          7720.8  
179              Rural          7843.1  
207              Rural          8085.0  
189              Rural          8093.7  
199              Rural          8134.8  
125              Rural          8154.9  
27               Rural          8199.3  
15               Rural          8366.9  
23               Rural          8950.1  
35               Rural          9421.3  
3                Rural          9494.7  
99               Rural          9514.3  
79               Rural          9561.7

In [50]:
total = chart15_df['Total expenses'].sum()
print('Wholesale trade:', float(chart15_df[chart15_df['North American Industry Classification System, NAICS'] == 'Wholesale trade'].iloc[0]['Total expenses'])/total)
print('Agriculture, forestry, fishing and hunting:', float(chart15_df[chart15_df['North American Industry Classification System, NAICS'] == 'Agriculture, forestry, fishing and hunting'].iloc[0]['Total expenses'])/total)
print('Retail trade:', float(chart15_df[chart15_df['North American Industry Classification System, NAICS'] == 'Retail trade'].iloc[0]['Total expenses'])/total)

Wholesale trade: 0.06667089677331185
Agriculture, forestry, fishing and hunting: 0.06620372565480658
Retail trade: 0.0663403906387275


**In 2019, rural area, 'Wholesale trade' (6.67%), 'Retail trade' (6.63%), 'Agriculture, forestry, fishing and hunting' (6.62%) are top 3 industries, which accounted for only 19.92% of rural medium business expenses. This might mean that the expenses differences between industries were not large.**

# Small businesses

### Total expenses - All industry - Small businesses

In [51]:
chart2_df = sb_revenue_value_3y[~(sb_revenue_value_3y['Total expenses, quality indicator'].isna()) & (sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_value_3y['Total expenses, quality indicator'] < 'E')]
chart2_df = chart2_df[(chart2_df['Total expenses, percent of businesses reporting'].astype(float) >= 90)]
chart2_df = chart2_df[['Reference year','Location indicator','Total expenses','Incorporation status']]
chart2_df['temp'] = chart2_df['Location indicator'] + ' - ' + chart2_df['Incorporation status']

In [52]:
chart2 = alt.Chart(chart2_df, title=alt.TitleParams(text = "Average total expenses of small businesses", 
                                           subtitle = "Canada, all inductries, by rural and urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('temp', title=''),
    y=alt.Y('Total expenses:Q', title='Total Expenses'),
    color=alt.Color('Incorporation status', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=150, height=300)

In [53]:
chart2_1_df = sb_revenue_persent_3y[~(sb_revenue_persent_3y['Total expenses (%), quality indicator'].isna()) & (sb_revenue_persent_3y['Geography - code'] == 0) & (sb_revenue_persent_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_persent_3y['Total expenses (%), quality indicator'] < 'E')]
chart2_1_df = chart2_1_df[(chart2_1_df['Total expenses (%), percent of businesses reporting'].astype(float) >= 90)]
chart2_1_df = chart2_1_df[['Reference year','Location indicator','Total expenses (%)','Incorporation status']]
chart2_1_df['temp'] = chart2_1_df['Location indicator'] + ' - ' + chart2_1_df['Incorporation status']

In [54]:
chart2_1 = alt.Chart(chart2_1_df, title=alt.TitleParams(text = "Percentage of total expenses in total revenues of medium businesses", 
                                           subtitle = "Canada, all inductries, by rural and urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('temp', title=''),
    y=alt.Y('Total expenses (%):Q', title='Total Expenses (%)'),
    color=alt.Color('Incorporation status', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=150, height=300)

In [55]:
chart2 | chart2_1

alt.HConcatChart(...)

In [56]:
analysis2 = pd.merge(left=chart2_df[chart2_df['Location indicator'] == 'Rural'][['Reference year','Location indicator','Total expenses','Incorporation status']],right=chart2_df[chart2_df['Location indicator'] == 'Urban'][['Reference year','Location indicator','Total expenses','Incorporation status']],how='inner',on=['Reference year','Incorporation status'])
analysis2

Reference year Location indicator_x Total expenses_x Incorporation status  \
0          2017.0                Rural               92       Unincorporated   
1          2017.0                Rural            592.4         Incorporated   
2          2017.0                Rural            408.7       All businesses   
3          2018.0                Rural             91.5       Unincorporated   
4          2018.0                Rural            596.6         Incorporated   
5          2018.0                Rural            412.5       All businesses   
6          2019.0                Rural             89.2       Unincorporated   
7          2019.0                Rural            606.8         Incorporated   
8          2019.0                Rural            417.2       All businesses   

  Location indicator_y Total expenses_y  
0                Urban             70.3  
1                Urban            500.6  
2                Urban            340.1  
3                Urban             68.8  
4                Urban            499.2  
5                Urban            340.1  
6                Urban             65.9  
7                Urban            504.9  
8                Urban            338.5

In [57]:
analysis2[analysis2['Incorporation status'] == 'All businesses']['Total expenses_x'].astype(float).mean()

412.8

In [58]:
analysis2[analysis2['Incorporation status'] == 'All businesses']['Total expenses_y'].astype(float).mean()

339.56666666666666

In [59]:
analysis2_1 = pd.merge(left=chart2_1_df[chart2_1_df['Location indicator'] == 'Rural'][['Reference year','Location indicator','Total expenses (%)','Incorporation status']],right=chart2_1_df[chart2_1_df['Location indicator'] == 'Urban'][['Reference year','Location indicator','Total expenses (%)','Incorporation status']],how='inner',on=['Reference year','Incorporation status'])
analysis2_1

Reference year Location indicator_x Total expenses (%)_x  \
0          2017.0                Rural                 70.5   
1          2017.0                Rural                 95.2   
2          2017.0                Rural                 92.5   
3          2018.0                Rural                 70.3   
4          2018.0                Rural                 95.7   
5          2018.0                Rural                   93   
6          2019.0                Rural                 69.5   
7          2019.0                Rural                 96.4   
8          2019.0                Rural                 93.6   

  Incorporation status Location indicator_y Total expenses (%)_y  
0       Unincorporated                Urban                 60.9  
1         Incorporated                Urban                 88.3  
2       All businesses                Urban                 85.4  
3       Unincorporated                Urban                 60.7  
4         Incorporated                Urban                 88.3  
5       All businesses                Urban                 85.4  
6       Unincorporated                Urban                 59.9  
7         Incorporated                Urban                 88.9  
8       All businesses                Urban                 85.9

**In the three years from 2017 to 2019, whatever incorporation status are, average total expenses of businesses in rural area always larger than urban area. Over these three years, the average total expenses for all rural small businesses was 412.8K, while for urban medium businesses, this value was about 339.6K. Compared with urban businesses, rural businesses have higher total expenses as a percentage of their total revenues. In recorded 3 years (2017,2018,2019), the total expenses percentage of all rural businesses increased 1.1%, while the total expenses percentage of all urban businesses decreased 0.5%.** 

**In 2019, the average total expenses for incorporated rural small businesses was 606.8K, while for unincorporated businesses, this value was 89.2K. The corresponding values were 504.9K for incorporated urban small businesses, and 65.9K for unincorporated urban small businesses. The value for incorporated businesses was much higher than unincorporated businesses, while the rural businesses had higher value than urban businesses.**

### Rural/Urban direct/indirect cost expenses - all industry - small businesses - all businesses

In [60]:
chart5_df = sb_revenue_value_3y[(sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['Incorporation status'] == 'All businesses') & (sb_revenue_value_3y['Location indicator'] == 'Rural') & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (sb_revenue_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart5_df = chart5_df[['Reference year','Location indicator','Cost of sales (direct expenses)','Operating expenses (indirect expenses)']]
chart5_cp = chart5_df

In [61]:
chart5_df = chart5_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [62]:
chart5 = alt.Chart(chart5_df,title=alt.TitleParams(text = "Average direct/indirect expenses of small businesses", 
                                           subtitle = "Canada, all inductries, rural areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('type', title=''),
    y=alt.Y('value:Q', title='Expenses Value'),
    color=alt.Color('type', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [63]:
chart8_df = sb_revenue_value_3y[(sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['Incorporation status'] == 'All businesses') & (sb_revenue_value_3y['Location indicator'] == 'Urban') & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (sb_revenue_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart8_df = chart8_df[['Reference year','Location indicator','Cost of sales (direct expenses)','Operating expenses (indirect expenses)']]
chart8_cp = chart8_df

In [64]:
chart8_df = chart8_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [65]:
chart8 = alt.Chart(chart8_df,title=alt.TitleParams(text = "Average direct/indirect expenses of small businesses", 
                                           subtitle = "Canada, all inductries, urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('type', title=''),
    y=alt.Y('value:Q', title='Expenses Value'),
    color=alt.Color('type', legend=None),
    column=alt.Column('Reference year', title='')
).properties(width=100, height=300)

In [66]:
chart5 | chart8

alt.HConcatChart(...)

In [67]:
chart5_cp

Reference year Location indicator Cost of sales (direct expenses)  \
3          2017.0              Rural                           144.1   
3          2018.0              Rural                           144.2   
3          2019.0              Rural                           143.9   

  Operating expenses (indirect expenses)  
3                                  264.6  
3                                  268.3  
3                                  273.3

In [68]:
chart5_cp['Cost of sales (direct expenses)'].astype(float).mean()

144.06666666666663

In [69]:
chart5_cp['Operating expenses (indirect expenses)'].astype(float).mean()

268.73333333333335

In [70]:
chart8_cp

Reference year Location indicator Cost of sales (direct expenses)  \
6          2017.0              Urban                           139.6   
6          2018.0              Urban                           137.4   
6          2019.0              Urban                           134.7   

  Operating expenses (indirect expenses)  
6                                  200.5  
6                                  202.7  
6                                  203.8

In [71]:
chart8_cp['Cost of sales (direct expenses)'].astype(float).mean()

137.23333333333332

In [72]:
chart8_cp['Operating expenses (indirect expenses)'].astype(float).mean()

202.33333333333334

**For small rural businesses, in recorded 3 years, the average direct expenses was 144.1K which was lower than the average indirect expenses (268.7K).For urban businesses, between 2017 and 2019, the average direct expenses (137.2K) was also lower than the average indirect expenses (202.3K). whatever expenses types are, average expenses values of businesses in rural area always larger than urban area.**
 

### Rural/Urban different type cost expenses - all industry - small businesses - all businesses

In [73]:
chart11_df = sb_revenue_value_3y[(sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['Incorporation status'] == 'All businesses') & (sb_revenue_value_3y['Location indicator'] == 'Rural') & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (sb_revenue_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart11_df = chart11_df[['Reference year','Location indicator',
                        'Wages and benefits', 'Purchases, materials and sub-contracts', 'Opening inventory', 
                           'Closing inventory', 'Labour and commissions', 'Amortization and depletion',
                           'Repairs and maintenance', 'Utilities and telephone/telecommunication',
                         'Rent', 'Interest and bank charges', 'Professional and business fees',
                         'Advertising and promotion', 'Delivery, shipping and warehouse', 'Insurance', 
                         'Other indirect expenses']]
chart11_cp = chart11_df

In [74]:
chart11_df = chart11_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [75]:
chart11_df['value'] = chart11_df['value'].astype(float)

In [76]:
chart11 = alt.Chart(chart11_df,title=alt.TitleParams(text = "Distribution of total expenses", 
                                           subtitle = "Canada, all inductries, small businesses, rural areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('value:Q', title='Expenses Value'),
    y=alt.Y('type', title='',sort='x'),
    color=alt.Color('type', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=350, height=250)

In [77]:
chart12_df = sb_revenue_value_3y[(sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['Incorporation status'] == 'All businesses') & (sb_revenue_value_3y['Location indicator'] == 'Urban') & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] == '0') & (sb_revenue_value_3y['Cost of sales (direct expenses), quality indicator'] < 'D') & (sb_revenue_value_3y['Operating expenses (indirect expenses)'] < 'D')]
chart12_df = chart12_df[['Reference year','Location indicator',
                        'Wages and benefits', 'Purchases, materials and sub-contracts', 'Opening inventory', 
                           'Closing inventory', 'Labour and commissions', 'Amortization and depletion',
                           'Repairs and maintenance', 'Utilities and telephone/telecommunication',
                         'Rent', 'Interest and bank charges', 'Professional and business fees',
                         'Advertising and promotion', 'Delivery, shipping and warehouse', 'Insurance', 
                         'Other indirect expenses']]
chart12_cp = chart12_df

In [78]:
chart12_df = chart12_df.melt(id_vars=['Reference year', 'Location indicator'], var_name='type',
             value_name='value')

In [79]:
chart12_df['value'] = chart12_df['value'].astype(float)

In [80]:
chart12 = alt.Chart(chart12_df,title=alt.TitleParams(text = "Distribution of total expenses", 
                                           subtitle = "Canada, all inductries, small businesses, urban areas, 2017 - 2019")).mark_bar().encode(
    x=alt.X('value:Q', title='Expenses Value'),
    y=alt.Y('type', title='',sort='x'),
    color=alt.Color('type', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=350, height=250)

In [81]:
chart11 | chart12

alt.HConcatChart(...)

In [82]:
chart11_cp

Reference year Location indicator Wages and benefits  \
3          2017.0              Rural               21.6   
3          2018.0              Rural               21.8   
3          2019.0              Rural               22.7   

  Purchases, materials and sub-contracts Opening inventory Closing inventory  \
3                                  123.1              26.3              26.9   
3                                  123.3              25.4              26.3   
3                                  121.8              25.6              26.2   

  Labour and commissions Amortization and depletion Repairs and maintenance  \
3                   67.4                       26.3                    16.5   
3                   67.9                       26.4                    16.6   
3                   68.7                       27.4                    16.9   

  Utilities and telephone/telecommunication  Rent Interest and bank charges  \
3                                        15  13.4                       8.4   
3                                        16  13.6                       9.1   
3                                      15.9  13.9                      10.2   

  Professional and business fees Advertising and promotion  \
3                            7.3                       4.2   
3                            7.3                       4.3   
3                            7.4                       4.3   

  Delivery, shipping and warehouse Insurance Other indirect expenses  
3                              1.9       7.1                      97  
3                              1.9       7.2                    97.9  
3                              1.8       7.6                    99.1

In [83]:
total = chart11_df[chart11_df['Reference year'] == 2019]['value'].astype(float).sum()
print('Purchases, materials and sub-contracts:', float(chart11_cp[chart11_cp['Reference year'] == 2019].iloc[0]['Purchases, materials and sub-contracts'])/total)

print('Other indirect expenses:', float(chart11_cp[chart11_cp['Reference year'] == 2019].iloc[0]['Other indirect expenses'])/total)

print('Labour and commissions:', float(chart11_cp[chart11_cp['Reference year'] == 2019].iloc[0]['Labour and commissions'])/total)

print('Delivery, shipping and warehouse:', float(chart11_cp[chart11_cp['Reference year'] == 2019].iloc[0]['Delivery, shipping and warehouse'])/total)

Purchases, materials and sub-contracts: 0.2594249201277955
Other indirect expenses: 0.21107561235356762
Labour and commissions: 0.1463258785942492
Delivery, shipping and warehouse: 0.0038338658146964857


In [84]:
chart12_cp

Reference year Location indicator Wages and benefits  \
6          2017.0              Urban               22.8   
6          2018.0              Urban               22.6   
6          2019.0              Urban               22.5   

  Purchases, materials and sub-contracts Opening inventory Closing inventory  \
6                                  118.4              24.8              26.4   
6                                    117              24.6              26.8   
6                                  113.3              25.4              26.5   

  Labour and commissions Amortization and depletion Repairs and maintenance  \
6                   73.7                       12.5                     8.9   
6                   74.3                       12.7                     9.1   
6                   74.7                       13.1                     9.1   

  Utilities and telephone/telecommunication  Rent Interest and bank charges  \
6                                         9  16.1                       7.5   
6                                       9.1  16.2                       8.4   
6                                       8.9  16.3                       8.7   

  Professional and business fees Advertising and promotion  \
6                           12.3                         6   
6                           12.3                         6   
6                           12.3                         6   

  Delivery, shipping and warehouse Insurance Other indirect expenses  
6                              1.4       4.8                    48.3  
6                              1.4       4.9                    48.4  
6                              1.4         5                    48.4

In [85]:
total = chart12_df[chart12_df['Reference year'] == 2019]['value'].astype(float).sum()
print('Purchases, materials and sub-contracts:', float(chart12_cp[chart12_cp['Reference year'] == 2019].iloc[0]['Purchases, materials and sub-contracts'])/total)
print('Other indirect expenses:', float(chart12_cp[chart12_cp['Reference year'] == 2019].iloc[0]['Other indirect expenses'])/total)
print('Labour and commissions:', float(chart12_cp[chart12_cp['Reference year'] == 2019].iloc[0]['Labour and commissions'])/total)
print('Delivery, shipping and warehouse:', float(chart12_cp[chart12_cp['Reference year'] == 2019].iloc[0]['Delivery, shipping and warehouse'])/total)

Purchases, materials and sub-contracts: 0.28932584269662925
Other indirect expenses: 0.12359550561797754
Labour and commissions: 0.19075587334014302
Delivery, shipping and warehouse: 0.003575076608784474


**The distribution of small business expenses among expenses types was similar in the three years from 2017 to 2019, but it was different in rural and urban area.**

**In 2019, rural area, 'Purchases, materials and sub-contracts' (25.94%), 'Other indirect expenses' (21.11%), 'Labour and commissions' (14.63%) were top 3 expenses types, which accounted for more than half (61.68%) of rural small business expenses, while 'Delivery, shipping and ware hourse' (0.38%) was the least expenses type. For urban area, 'Purchases, materials and sub-contracts' (28.93%), 'Labour and commissions' (19.08%), 'Other indirect expenses' (12.36%) were still top 3 expenses types that accounted 60.37% expenses in total, while 'Delivery, shipping and ware hourse' (0.38%) was the least expenses type. Compared to the rural businesses, urban businesses speend more expenses on 'Labour and commissions'.**

### industry shares of total expenses - rural area

In [86]:
chart16_df = sb_revenue_value_3y[(sb_revenue_value_3y['Geography - code'] == 0) & (sb_revenue_value_3y['Incorporation status'] == 'All businesses') & (sb_revenue_value_3y['Location indicator'] == 'Rural') & (sb_revenue_value_3y['North American Industry Classification System, NAICS - code'] != '0') & (sb_revenue_value_3y['Total expenses, quality indicator'] < 'E')]
chart16_df = chart16_df[['North American Industry Classification System, NAICS','Reference year','Location indicator','Total expenses']]
chart16_df = chart16_df[(chart16_df['Location indicator'] == 'Rural') & (chart16_df['Reference year'] == 2019)]

In [87]:
chart16_df.loc[:,'Total expenses'] = chart16_df.loc[:,'Total expenses'].astype(float)

In [88]:
alt.Chart(chart16_df,title=alt.TitleParams(text = "Industry shares of total expenses", 
                                           subtitle = "Canada, small businesses, urban areas, 2019")).mark_bar().encode(
    x=alt.X('Total expenses:Q', title='Total Expenses'),
    y=alt.Y('North American Industry Classification System, NAICS', title='',sort='x'),
    color=alt.Color('North American Industry Classification System, NAICS', legend=None),
    row=alt.Row('Reference year', title='')
).properties(width=400, height=300)

alt.Chart(...)

In [89]:
chart16_df.sort_values(['Total expenses'])

North American Industry Classification System, NAICS  Reference year  \
8022                  Real estate and rental and leasing            2019.0   
8418     Professional, scientific and technical services            2019.0   
9111                                Educational services            2019.0   
9309                   Health care and social assistance            2019.0   
8616             Management of companies and enterprises            2019.0   
9804                  Arts, entertainment and recreation            2019.0   
8814   Administrative and support, waste management a...            2019.0   
7329                 Information and cultural industries            2019.0   
10497      Other services (except public administration)            2019.0   
696        Mining, quarrying, and oil and gas extraction            2019.0   
6141                      Transportation and warehousing            2019.0   
1290                                        Construction            2019.0   
10200                    Accommodation and food services            2019.0   
1092                                           Utilities            2019.0   
1686                                       Manufacturing            2019.0   
3864                                     Wholesale trade            2019.0   
102           Agriculture, forestry, fishing and hunting            2019.0   
4854                                        Retail trade            2019.0   

      Location indicator  Total expenses  
8022               Rural           129.4  
8418               Rural           170.4  
9111               Rural           183.5  
9309               Rural           216.3  
8616               Rural           221.1  
9804               Rural           249.5  
8814               Rural           250.6  
7329               Rural           262.7  
10497              Rural           264.8  
696                Rural           365.3  
6141               Rural           388.5  
1290               Rural           404.3  
10200              Rural           475.9  
1092               Rural           520.8  
1686               Rural           616.0  
3864               Rural           729.5  
102                Rural           735.1  
4854               Rural           814.6

In [90]:
total = chart16_df['Total expenses'].sum()
print('Wholesale trade:', float(chart16_df[chart16_df['North American Industry Classification System, NAICS'] == 'Wholesale trade'].iloc[0]['Total expenses'])/total)
print('Agriculture, forestry, fishing and hunting:', float(chart16_df[chart16_df['North American Industry Classification System, NAICS'] == 'Agriculture, forestry, fishing and hunting'].iloc[0]['Total expenses'])/total)
print('Retail trade:', float(chart16_df[chart16_df['North American Industry Classification System, NAICS'] == 'Retail trade'].iloc[0]['Total expenses'])/total)

Wholesale trade: 0.1042396010459683
Agriculture, forestry, fishing and hunting: 0.10503979537887773
Retail trade: 0.11639969706928825


**In 2019, rural area, 'Retail trade' (11.64%), 'Agriculture, forestry, fishing and hunting' (10.50%), 'Wholesale trade' (10.42%) are top 3 industries, which accounted for only 32.56% of rural small business expenses. This might mean that the expenses differences between industries were not large.**